lib/options_test.go

```diff
@@ -318,6 +318,28 @@ func TestOptions(t *testing.T) {
 		assert.Equal(t, net.IPv4zero, opts.BlacklistIPs[0].IP)
 		assert.Equal(t, net.CIDRMask(1, 1), opts.BlacklistIPs[0].Mask)
 	})
+	t.Run("BlockedHostnames", func(t *testing.T) {
+		hostnames := HostnameTrie{}
+		assert.NoError(t, hostnames.Insert("test.k6.io"))
+		assert.Error(t, hostnames.Insert("inval*d.pattern"))
+		assert.NoError(t, hostnames.Insert("*valid.pattern"))
+		opts := Options{}.Apply(Options{
+			BlockedHostnames: &hostnames,
+		})
+		assert.NotNil(t, opts.BlockedHostnames)
+		assert.NotEmpty(t, opts.BlockedHostnames)
+		matches, _ := opts.BlockedHostnames.Contains("k6.io")
+		assert.False(t, matches)
+		matches, _ = opts.BlockedHostnames.Contains("test.k6.io")
+		assert.True(t, matches)
```

Comment:
Please add a test that this will block an uppercase version of the domain as well